thirdparty_containers/system-test/dockerfile/system-test.sh

```diff
@@ -0,0 +1,92 @@
+#/bin/bash
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#Set up Java to be used by the system-test
+
+if [ -d /java/jre/bin ];then
+	echo "Using mounted Java8"
+	export JAVA_BIN=/java/jre/bin
+	export JAVA_HOME=/java
+	export PATH=$JAVA_BIN:$PATH
+	java -version
+elif [ -d /java/bin ]; then
+	echo "Using mounted Java9"
+	export JAVA_BIN=/java/bin
+	export JAVA_HOME=/java
+	export PATH=$JAVA_BIN:$PATH
+	java -version
+else
+	echo "Using docker image default Java"
+	java_path=$(type -p java)
+	suffix="/java"
+	java_root=${java_path%$suffix}
+	export JAVA_BIN="$java_root"
+	echo "JAVA_BIN is: $JAVA_BIN"
+	$JAVA_BIN/java -version
+fi
+
+TEST_SUITE=$1
+
+echo "PATH is : $PATH"
+echo "JAVA_HOME is : $JAVA_HOME"
+echo "type -p java is :"
+type -p java
+echo "java -version is: \n"
+java -version
+
+cd /openjdk-tests
+
+./get.sh -t /openjdk-tests -p x64_linux -i openj9
+
+cd TestConfig
+
+export SPEC=linux_x86-64_cmprssptrs
+export BUILD_LIST=systemtest
+
+# Generates test targets applicable to the platform, implementation and version
+echo "Generating test targets applicable to the platform, implementation and version..." 
+make -f run_configure.mk
+
+# fetches test material and compiles it, based on build.xml files in the test directories
+echo "Building test material..." 
+make compile
+
+testsuite=$@
+
+#Run tests
+if [ "$testsuite" == "mauve" ]; then
+echo "Running Mauve tests..."
+
+	make _MauveSingleThreadLoadTest
+	make _MauveSingleInvocationLoadTest
+	make _MauveMultiThreadLoadTest
+
+elif [ "$testsuite" == "sharedclasses" ]; then
+	echo "Running the Shared classes system tests" 
+
+	make _SharedClassesAPI
+	make _SharedClassesWorkload
+	make _SharedClassesWorkloadTest_Softmx_Increase
+	make _SharedClassesWorkloadTest_Softmx_IncreaseDecrease
+	make _SharedClassesWorkloadTest_Softmx_Increase_JitAot_Linux
+	make _SharedClasses.SCM01.SingleCL
+	make _SharedClasses.SCM01.MultiCL
+	make _SharedClasses.SCM01.MultiThread
+	make _SharedClasses.SCM01.MultiThreadMultiCL
+	make _SharedClasses.SCM23.SingleCL
+	make _SharedClasses.SCM23.MultiCL
+	make _SharedClasses.SCM23.MultiThread
+	make _SharedClasses.SCM23.MultiThreadMultiCL
+fi
```

Comment:
Could we make this more generic? Instead of passing `TEST_SUITE` and redefine test groups in the file, can we just reuse https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/maketest.sh? In this way, we can pass in any existing test targets.

https://github.com/AdoptOpenJDK/openjdk-tests/blob/ac0012d53f1223456c4b8dfc9a5a99675aa279ba/buildenv/jenkins/JenkinsfileBase#L31